rpc/src/v1/impls/eth_pubsub.rs

```diff
@@ -50,6 +52,70 @@ pub struct EthPubSubClient<C> {
 	heads_subscribers: Arc<RwLock<Subscribers<Client>>>,
 	logs_subscribers: Arc<RwLock<Subscribers<(Client, EthFilter)>>>,
 	transactions_subscribers: Arc<RwLock<Subscribers<Client>>>,
+	sync_subscribers: Arc<RwLock<Subscribers<Client>>>,
+}
+
+impl<C> EthPubSubClient<C>
+	where
+		C: 'static + BlockChainClient
+{
+	/// adds a sync notification channel to the pubsub client
+	pub fn add_sync_notifier(&mut self, receiver: Notification<SyncState>, sync: Arc<SyncProvider>) {
+		let handler = self.handler.clone();
+		let client = self.handler.client.clone() as Arc<BlockChainClient>;
+
+		self.handler.executor.spawn(
+			receiver.for_each(move |state| {
+				let queue_info = client.queue_info();
+				let status = sync.status();
+
+				let is_syncing_state = match state { SyncState::Idle | SyncState::NewBlocks => false, _ => true };
+				let is_verifying = queue_info.unverified_queue_size + queue_info.verified_queue_size > 3;
+
+				let highest_block = U64::from(status.highest_block_number.unwrap_or(status.start_block_number));
+
+				let sync_state = pubsub::PubSubSyncStatus {
+					is_syncing: is_verifying || is_syncing_state,
+					starting_block: status.start_block_number.into(),
+					current_block: client.chain_info().best_block_number.into(),
+					highest_block: highest_block.into(),
+				};
+				handler.notify_syncing(sync_state);
+				Ok(())
+			})
+		)
+	}
+}
+
+impl<C> EthPubSubClient<LightFetch<C>>
+	where
+		C: 'static + LightSyncProvider + LightNetworkDispatcher + ManageNetwork
+{
+	/// adds a sync notification channel to the pubsub client
+	pub fn add_light_sync_notifier(&mut self, receiver: Notification<SyncState>, sync: Arc<LightSyncInfo>) {
```

Comment:
I thought that we extract `QueueInfo` trait to unify the `add_light_sync_notifier` and `add_sync_notifier`, if we don't do that, then the other changes are just unnecessary.
Can we unify this completely the logic seems to be exactly the same?

If there is something that I'm missing we can also pass a closure that produces `PubSubSyncStatus` and then the mode-specific logic can be there.